.ci/component_descriptor-callback

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import argparse
+import dataclasses
+import json
+
+try:
+    import yaml
+    _have_yaml = True
+except ImportError:
+    _have_yaml = False
+
+import ocm
+
+
+if _have_yaml:
+    _yaml_or_json_load = yaml.safe_load
+else:
+    _yaml_or_json_load = json.load
+
+
+def _parse_yaml_or_json(path) -> dict | list:
+    with open(path) as f:
+        return _yaml_or_json_load(f)
+
+
+def main(parsed):
+    raw = _parse_yaml_or_json(parsed.file)
+    component_references = raw['component'].get('componentReferences', [])
+
+    component_references.append(dataclasses.asdict(ocm.ComponentReference(
+        name='bdba-client',
+        componentName='ocm.software/ocm-gear/bdba-client',
+        version='0.2.0',
```

Comment:
this is not a nice place to maintain dependency-versions. I suggest to put this into a dedicated file (either a plain text file containing just the version, or a yaml file, as we already have been doing for the past years in other repositories..)

Also, I think it would be preferable to use a shellscript - this should mean a lot less boilerplate.

I think it should be preferable to keep contract towards component-descriptor-callback-scripts as small as possible. While it is obviously not a strong hurdle for us to offer python + gha-libs, writing such callbacks like this will make it a lot more difficult to locally run the script, esp. considering other repositories.